addons/docker/Dockerfile.tt

```diff
@@ -1,16 +1,19 @@
 FROM ruby:<%= RUBY_VERSION %>-slim
 
+ARG BUILD_ENV=development
 ARG RUBY_ENV=development
+<%- if WEB_VARIANT -%>
 ARG NODE_ENV=development
-ARG BUILD_ENV=development
 ARG ASSET_HOST=http://localhost
+<%- end -%>
 
 # Define all the envs here
-ENV RACK_ENV=$RUBY_ENV \
-    RAILS_ENV=$RUBY_ENV \
+ENV BUILD_ENV=$BUILD_ENV \
+    RACK_ENV=$RUBY_ENV \
+    RAILS_ENV=$RUBY_ENV<%- if WEB_VARIANT -%> \
```

Comment:
The `if` placement is a bit awkward but I think it's fine as it prevents to create another `RUN` line (thus another Docker image layer).